services/search/src/search/routes/filter.py

```diff
@@ -0,0 +1,278 @@
+                        features,
+                    )
+                with StepProfiler(method="filter_endpoint", step="execute filter query"):
+                    num_rows_total, pa_table = execute_filter_query(
+                        index_file_location=index_file_location,
+                        columns=supported_columns,
+                        where=where,
+                        limit=length,
+                        offset=offset,
+                    )
+                with StepProfiler(method="search_endpoint", step="clean cache randomly"):
+                    clean_cached_assets_randomly(
+                        clean_cache_proba=clean_cache_proba,
+                        dataset=dataset,
+                        cached_assets_directory=cached_assets_directory,
+                        keep_first_rows_number=keep_first_rows_number,
+                        keep_most_recent_rows_number=keep_most_recent_rows_number,
+                        max_cleaned_rows_number=max_cleaned_rows_number,
+                    )
+                with StepProfiler(method="filter_endpoint", step="create response"):
+                    response = create_response(
+                        dataset=dataset,
+                        config=config,
+                        split=split,
+                        cached_assets_base_url=cached_assets_base_url,
+                        cached_assets_directory=cached_assets_directory,
+                        s3_client=s3_client,
+                        cached_assets_s3_bucket=cached_assets_s3_bucket,
+                        cached_assets_s3_folder_name=cached_assets_s3_folder_name,
+                        pa_table=pa_table,
+                        offset=offset,
+                        features=features,
+                        unsupported_columns=unsupported_columns,
+                        num_rows_total=num_rows_total,
+                    )
+                with StepProfiler(method="filter_endpoint", step="generate the OK response"):
+                    return get_json_ok_response(content=response, max_age=max_age_long, revision=revision)
+            except Exception as e:
+                error = e if isinstance(e, ApiError) else UnexpectedError("Unexpected error.", e)
+                with StepProfiler(method="filter_endpoint", step="generate API error response"):
+                    return get_json_api_error_response(error=error, max_age=max_age_short, revision=revision)
+
+    return filter_endpoint
+
+
+def execute_filter_query(
+    index_file_location: str, columns: list[str], where: str, limit: int, offset: int
+) -> tuple[int, pa.Table]:
+    con = duckdb.connect(database=index_file_location, read_only=True)
+    # TODO: Address possible SQL injection CWE-89
+    filter_query = FILTER_QUERY.format(columns=",".join(columns), where=where, limit=limit, offset=offset)
+    pa_table = con.sql(filter_query).arrow()
+    filter_count_query = FILTER_COUNT_QUERY.format(where=where)
```

Comment:
same here

Reply:
If the `where` is problematic, it raises an exception in the precedent SQL query.

Reply:
At the end, I put the count query under try-except as well.